lib/service/connect.go

```diff
@@ -104,19 +111,22 @@ func (process *TeleportProcess) reconnectToAuthService(role types.SystemRole) (*
 	}
 }
 
-func (process *TeleportProcess) compareVersions(resp *proto.PingResponse) error {
+func (process *TeleportProcess) authServerTooOld(resp *proto.PingResponse) (bool, error) {
```

Comment:
nit: this can be made simpler by returning a `trace.NotImplemented` instead of an extra `bool` return. Then, the caller can check `trace.IsNotImplemented(err)` instead. Something like:

```
if err := process.authServerTooOld(...); trace.IsNotImplemented(err) {
    if !process.Config.SkipVersionCheck {
        return nil, trace.Wrap(err)
    }
} else if err != nil {
    return nil, trace.Wrap(err)
}
```